ec2/instance.py

```diff
@@ -622,3 +636,68 @@ def _get_instance_subnet(provider_variables):
             'instance may only be attached to one subnet')
 
     return list_of_subnets[0] if list_of_subnets else None
+
+
+def _get_network_interface_object(ec2_client, eni_id):
+    '''Gets a NetworkInterface object from AWS'''
+    eni_objs = ec2_client.get_all_network_interfaces(
+        network_interface_ids=[eni_id])
+    if not isinstance(eni_objs, list) or len(eni_objs) < 1:
+        raise NonRecoverableError(
+            'Could not get NetworkInterface object of {0}'
+            .format(eni_id))
+    return eni_objs[0]
+
+
+def _find_attached_eni():
```

Comment:
use the get_target_external_resource_ids util from utils. if it's got issues refactor it.